src/arch/ia32/descriptor/tss/permissions.rs

```diff
@@ -0,0 +1,139 @@
+//! A module containing the implementation and tests for the [`Permissions`]
+//! structure.
+use crate::arch::ia32::PrivilegeLevel;
+use bit_field::BitField;
+use core::fmt;
+
+/// A structure representing the bits 8 to 15 from a
+/// [TssDescriptor](super::TssDescriptor) structure.
+///
+/// It gathers the following fields:
+/// - Segment type (TYPE)
+/// - Descriptor privilege level (DPL)
+/// - Present bit (P)
+#[derive(Copy, Clone)]
+pub struct Permissions(u8);
+
+/// The set of all field offsets for the [`Permissions`] structure.
+mod offset {
+    /// Offset of the busy bit (B) within the [Permissions](super::Permissions)
+    /// structure
+    pub const BUSY: usize = 1;
+
+    /// Bounds of the descriptor privilege level (DPL) bits within the
+    /// [Permissions](super::Permissions) structure.
+    pub mod privilege_level {
+        /// Lower bit offset.
+        pub const LOWER: usize = 5;
+
+        /// Upper bit offset.
+        pub const UPPER: usize = 6;
+    }
+
+    /// Offset of the present bit (P) within the
+    /// [Permissions](super::Permissions) structure.
+    pub const PRESENT: usize = 7;
+}
+
+impl Default for Permissions {
+    fn default() -> Self {
+        Permissions(0b00001001)
+    }
+}
+
+impl Permissions {
+    /// Change a [`Permissions`]'s busy bit.
+    ///
+    /// # Arguments
+    ///
+    /// * `busy` - The desired bit value, `true` for bit value 1 and `false`
+    /// for bit value 0.
+    pub fn busy(self, busy: bool) -> Self {
+        use offset::BUSY;
+        Self(*self.0.clone().set_bit(BUSY, busy))
+    }
+
+    /// Change a [`Permissions`]'s privilege level by another one.
+    ///
+    /// # Arguments
+    ///
+    /// * `level` - The desired [`PrivilegeLevel`] value.
+    pub fn privilege_level(self, level: PrivilegeLevel) -> Self {
+        use offset::privilege_level::{LOWER, UPPER};
+        Self(*self.0.clone().set_bits(LOWER..=UPPER, level.into()))
+    }
+
+    /// Change a [`Permissions`]'s present bit.
+    ///
+    /// # Arguments
+    ///
+    /// * `present` - The desired bit value, `true` for bit value 1 and `false`
+    /// for bit value 0.
+    pub fn present(self, present: bool) -> Self {
+        use offset::PRESENT;
+        Self(*self.0.clone().set_bit(PRESENT, present))
+    }
+}
+
+impl fmt::Display for Permissions {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        use offset::{privilege_level, BUSY, PRESENT};
+        write!(
+            f,
+            "Busy: {}\nDPL: {}\nPresent {}",
```

Comment:
Here too if you want to put a semicolon after "Present"

Reply:
Done in 57418d7